### PR TITLE
Promote int to float for tan operation (consistent with Pytorch)

### PR DESCRIPTION
**Pull request by steventk-g**

### CODE REVIEW

File: torch_xla/csrc/ops/ops_lower_fn.cpp [hunk 1]
```diff
@@ -598,6 +599,11 @@ torch_xla::XlaOpVector Take::Lower(LoweringContext* loctx) const {
 
 torch_xla::XlaOpVector Tan::Lower(LoweringContext* loctx) const {
   xla::XlaOp xla_input = loctx->GetOutputOp(operand(0));
+  if (xla::primitive_util::IsIntegralType(XlaHelpers::TypeOfXlaOp(xla_input))) {
```
Comment (JackCaoG) [inline torch_xla/csrc/ops/ops_lower_fn.cpp:602]: I think we also need to update https://github.com/pytorch/xla/blob/9794392a519119a4e4b6ddd35f737ed9dd64eece/torch_xla/csrc/ops/ops_xla_shape_fn.cpp#L680-L682. I don't think result will be int type.

Comment (steventk-g) [reply torch_xla/csrc/ops/ops_lower_fn.cpp:602]: Done

File: test/cpp/test_aten_xla_tensor.cpp [hunk 2]
```diff
@@ -2833,6 +2833,18 @@ TEST_F(AtenXlaTensorTest, TestTan) {
...
+// torch_xla
+TEST_F(AtenXlaTensorTest, TestTanWithInt) {
+  torch::Tensor a = torch::rand({2, 2});
+  torch::Tensor b = torch::tan(a);
+  ForEachDevice([&](const torch::Device& device) {
+    torch::Tensor xla_a = CopyToDevice(a, device);
+    torch::Tensor xla_b = torch::tan(xla_a);
+    AllClose(b, xla_b, /*rtol=*/1e-3, /*atol=*/1e-5);
+  });
+}
```
Comment (JackCaoG) [inline test/cpp/test_aten_xla_tensor.cpp:2846]: maybe also check counters like other ccp test?

Comment (steventk-g) [reply test/cpp/test_aten_xla_tensor.cpp:2850]: Done

**Review by JackCaoG — APPROVED** [review]
Thanks!

**Pull request merged**